plugins/assistant/api/assistantJobGeneral.js

```diff
@@ -35,132 +37,136 @@ else {
+                                        log.w('Assistant plugin, feed reader returned undefined! Probably timeout. url: [%j] error: [%j] ', feedItem.url, parserErr);
+                                    }
+                                    //go through collected feed items
+                                    async.each(arrayForDataToNotify, function(feedMiddleItem, callbackMiddle) {
+                                    //go through all apps
+                                        async.each(providedInfo.appsData, function(ret_app_data, callbackInner) {
+                                            let apc = assistant.preparePluginSpecificFields(providedInfo, ret_app_data, PLUGIN_NAME);
+                                            let anc = assistant.prepareNotificationSpecificFields(apc, feedItem.anc_i18n, feedItem.anc_type, feedItem.anc_subtype, feedItem.anc_version);
+
+                                            assistant.createNotificationIfRequirementsMet(-1, feedItem.targetHour, (feedMiddleItem.blog_post_ready), feedMiddleItem.data, anc);
+                                            callbackInner(null);
+                                        }, function(err) {
+                                            if (err !== null) {
+                                                log.w('Assistant feed generation internal resolving with error:[%j]', err);
+                                            }
+                                            callbackMiddle(null);
+                                        });
+
+                                    }, function(feedHandlerErr) {
+                                        if (feedHandlerErr !== null) {
+                                            log.w('Assistant feed generation middle resolving with error:[%j]', feedHandlerErr);
+                                        }
+                                        callbackOuter(null);
+                                    });
                                 }
-                                callbackOuter(null);
                             });
-                        }
-                    });
 
-                }, function(err) {
-                    if (err !== null) {
-                        log.w('Assistant feed generation outer resolving with an error:[%j]', err);
-                    }
-                    else {
-                        log.i('Assistant for [%j] plugin resolving with no errors', PLUGIN_NAME);
+                        }, function(err) {
+                            if (err !== null) {
+                                log.w('Assistant feed generation outer resolving with an error:[%j]', err);
+                            }
+                            else {
+                                log.i('Assistant for [%j] plugin resolving with no errors', PLUGIN_NAME);
+                            }
+                            resolve();
+                        });
                     }
```

Comment:
I think here in else branch we need to resolve the Promise too, or else it will never be resolved